src/memo/memo.ml

```diff
@@ -575,7 +583,12 @@ module Stack_frame = struct
 end
 
 let handle_code_error ~frame (value : 'a Value.t) : 'a Value.t =
-  Result.map_error value ~f:(fun exn ->
+  Result.map_error value ~f:(fun exns ->
+      let exn =
+        match exns with
+        | [ exn ] -> exn
+        | _ -> assert false
```

Comment:
Why is this `assert false` correct? i.e. is this code unreachable and why?

Reply:
Just noticed `get_sync_exn` does the same. I'm guessing that the invariant is that in sync mode, the list is always of length 1. This invariant should be written down in a comment

Reply:
I would also arrange the code so that there is a single `assert false` for this invariant, so that it can be documented once and for all.